### DISCUSSION TITLE
failed to run custom build command for `ring v0.16.20`

### DISCUSSION THREAD

**Discussion by IkramulMurad** [shadowsocks/shadowsocks-android#2819] [Q&A]
Executing tasks: [:mobile:assembleDebug] in project /home/ikramulmurad/Desktop/Workspace/_Projects/Android-Projects/shadowsocks-android

> Task :buildSrc:compileKotlin UP-TO-DATE
> Task :buildSrc:compileJava NO-SOURCE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:pluginDescriptors UP-TO-DATE
> Task :buildSrc:processResources NO-SOURCE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:inspectClassesForKotlinIC UP-TO-DATE
> Task :buildSrc:jar UP-TO-DATE
> Task :buildSrc:assemble UP-TO-DATE
> Task :buildSrc:compileTestKotlin NO-SOURCE
> Task :buildSrc:pluginUnderTestMetadata UP-TO-DATE
> Task :buildSrc:compileTestJava NO-SOURCE
> Task :buildSrc:compileTestGroovy NO-SOURCE
> Task :buildSrc:processTestResources NO-SOURCE
> Task :buildSrc:testClasses UP-TO-DATE
> Task :buildSrc:test NO-SOURCE
> Task :buildSrc:validatePlugins UP-TO-DATE
> Task :buildSrc:check UP-TO-DATE
> Task :buildSrc:build UP-TO-DATE

> Configure project :core
AGPBI: {"kind":"warning","text":"The option setting 'android.enableR8.fullMode=true' is experimental.\nThe current default is 'false'.","sources":[{}]}
AGPBI: {"kind":"warning","text":"The option setting 'android.enableResourceOptimizations=false' is deprecated.\nThe current default is 'true'.\nIt will be removed in version 8.0 of the Android Gradle plugin.","sources":[{}]}

> Task :core:preBuild UP-TO-DATE
> Task :core:preDebugBuild UP-TO-DATE
> Task :plugin:preBuild UP-TO-DATE
> Task :plugin:preDebugBuild UP-TO-DATE
> Task :plugin:compileDebugAidl NO-SOURCE
> Task :core:compileDebugAidl UP-TO-DATE
> Task :plugin:packageDebugRenderscript NO-SOURCE
> Task :core:compileDebugRenderscript NO-SOURCE
> Task :core:generateDebugBuildConfig UP-TO-DATE
> Task :core:generateDebugResValues UP-TO-DATE
> Task :core:generateDebugResources UP-TO-DATE
> Task :core:packageDebugResources UP-TO-DATE
> Task :core:parseDebugLocalResources UP-TO-DATE
> Task :core:processDebugManifest UP-TO-DATE
> Task :plugin:compileDebugRenderscript NO-SOURCE
> Task :plugin:generateDebugResValues UP-TO-DATE
> Task :plugin:generateDebugResources UP-TO-DATE
> Task :plugin:packageDebugResources UP-TO-DATE
> Task :plugin:parseDebugLocalResources UP-TO-DATE
> Task :plugin:processDebugManifest UP-TO-DATE
> Task :plugin:generateDebugRFile UP-TO-DATE
> Task :core:generateDebugRFile UP-TO-DATE
> Task :plugin:generateDebugBuildConfig UP-TO-DATE
> Task :plugin:compileDebugKotlin UP-TO-DATE
> Task :plugin:javaPreCompileDebug UP-TO-DATE
> Task :plugin:compileDebugJavaWithJavac UP-TO-DATE
> Task :plugin:bundleLibCompileToJarDebug UP-TO-DATE
> Task :core:kaptGenerateStubsDebugKotlin UP-TO-DATE
> Task :core:kaptDebugKotlin UP-TO-DATE
> Task :core:compileDebugKotlin UP-TO-DATE
> Task :core:javaPreCompileDebug UP-TO-DATE
> Task :core:compileDebugJavaWithJavac UP-TO-DATE
> Task :core:bundleLibCompileToJarDebug UP-TO-DATE
> Task :mobile:getDependencies UP-TO-DATE
> Task :mobile:generateLicenses UP-TO-DATE
> Task :mobile:preBuild UP-TO-DATE
> Task :mobile:preDebugBuild UP-TO-DATE
> Task :mobile:compileDebugAidl NO-SOURCE
> Task :core:packageDebugRenderscript NO-SOURCE
> Task :mobile:compileDebugRenderscript NO-SOURCE
> Task :mobile:generateDebugBuildConfig UP-TO-DATE
> Task :core:compileDebugLibraryResources UP-TO-DATE
> Task :core:writeDebugAarMetadata UP-TO-DATE
> Task :plugin:writeDebugAarMetadata UP-TO-DATE
> Task :mobile:checkDebugAarMetadata UP-TO-DATE
> Task :mobile:generateDebugResValues UP-TO-DATE
> Task :mobile:generateDebugResources UP-TO-DATE
> Task :mobile:injectCrashlyticsMappingFileIdDebug UP-TO-DATE
> Task :mobile:processDebugGoogleServices UP-TO-DATE
> Task :mobile:mergeDebugResources UP-TO-DATE
> Task :mobile:createDebugCompatibleScreenManifests UP-TO-DATE
> Task :core:extractDeepLinksDebug UP-TO-DATE
> Task :mobile:extractDeepLinksDebug UP-TO-DATE
> Task :plugin:extractDeepLinksDebug UP-TO-DATE
> Task :mobile:processDebugMainManifest UP-TO-DATE
> Task :mobile:processDebugManifest UP-TO-DATE
> Task :mobile:processDebugManifestForPackage UP-TO-DATE
> Task :plugin:compileDebugLibraryResources UP-TO-DATE
> Task :mobile:processDebugResources UP-TO-DATE
> Task :mobile:compileDebugKotlin UP-TO-DATE
> Task :mobile:javaPreCompileDebug UP-TO-DATE
> Task :mobile:compileDebugJavaWithJavac UP-TO-DATE
> Task :mobile:compileDebugSources UP-TO-DATE
> Task :mobile:mergeDebugNativeDebugMetadata NO-SOURCE
> Task :core:mergeDebugShaders UP-TO-DATE
> Task :core:compileDebugShaders NO-SOURCE
> Task :core:generateDebugAssets UP-TO-DATE
> Task :core:packageDebugAssets UP-TO-DATE
> Task :mobile:mergeDebugShaders UP-TO-DATE
> Task :mobile:compileDebugShaders NO-SOURCE
> Task :mobile:generateDebugAssets UP-TO-DATE
> Task :plugin:mergeDebugShaders UP-TO-DATE
> Task :plugin:compileDebugShaders NO-SOURCE
> Task :plugin:generateDebugAssets UP-TO-DATE
> Task :plugin:packageDebugAssets UP-TO-DATE
> Task :mobile:mergeDebugAssets UP-TO-DATE
> Task :mobile:compressDebugAssets UP-TO-DATE
> Task :mobile:l8DexDesugarLibDebug UP-TO-DATE
> Task :core:processDebugJavaRes NO-SOURCE
> Task :core:bundleLibResDebug UP-TO-DATE
> Task :mobile:processDebugJavaRes NO-SOURCE
> Task :plugin:processDebugJavaRes NO-SOURCE
> Task :plugin:bundleLibResDebug UP-TO-DATE
> Task :mobile:mergeDebugJavaResource UP-TO-DATE
> Task :mobile:checkDebugDuplicateClasses UP-TO-DATE
> Task :mobile:desugarDebugFileDependencies UP-TO-DATE
> Task :mobile:mergeExtDexDebug UP-TO-DATE
> Task :core:bundleLibRuntimeToDirDebug UP-TO-DATE
> Task :plugin:bundleLibRuntimeToDirDebug UP-TO-DATE
> Task :mobile:dexBuilderDebug UP-TO-DATE
> Task :mobile:mergeProjectDexDebug UP-TO-DATE
> Task :core:configureNdkBuildDebug
> Task :mobile:mergeLibDexDebug UP-TO-DATE
> Task :core:buildNdkBuildDebug
> Task :plugin:mergeDebugJniLibFolders UP-TO-DATE
> Task :plugin:mergeDebugNativeLibs NO-SOURCE
> Task :plugin:stripDebugDebugSymbols NO-SOURCE
> Task :plugin:copyDebugJniLibsProjectOnly UP-TO-DATE
> Task :core:externalNativeBuildDebug
> Task :generateLinkerWrapper UP-TO-DATE

> Task :core:cargoBuildArm FAILED
   Compiling ring v0.16.20
   Compiling quote v1.0.9
   Compiling getrandom v0.2.3
   Compiling parking_lot_core v0.8.3
error: failed to run custom build command for `ring v0.16.20`

Caused by:
  process didn't exit successfully: `/home/ikramulmurad/Desktop/Workspace/_Projects/Android-Projects/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/debug/build/ring-120afe4acc70546b/build-script-build` (exit status: 101)
  --- stdout
  OPT_LEVEL = Some("0")
  TARGET = Some("armv7-linux-androideabi")
  HOST = Some("x86_64-unknown-linux-gnu")
  CC_armv7-linux-androideabi = None
  CC_armv7_linux_androideabi = None
  TARGET_CC = None
  CC = None
  CFLAGS_armv7-linux-androideabi = None
  CFLAGS_armv7_linux_androideabi = None
  TARGET_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("true")

  --- stderr
  running "arm-linux-androideabi-clang" "-O0" "-DANDROID" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-I" "include" "-Wall" "-Wextra" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-g3" "-DNDEBUG" "-c" "-o/home/ikramulmurad/Desktop/Workspace/_Projects/Android-Projects/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/armv7-linux-androideabi/debug/build/ring-52c596bac6120bfc/out/aesv8-armx-linux32.o" "/home/ikramulmurad/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/pregenerated/aesv8-armx-linux32.S"
  thread 'main' panicked at 'failed to execute ["arm-linux-androideabi-clang" "-O0" "-DANDROID" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-I" "include" "-Wall" "-Wextra" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-g3" "-DNDEBUG" "-c" "-o/home/ikramulmurad/Desktop/Workspace/_Projects/Android-Projects/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/armv7-linux-androideabi/debug/build/ring-52c596bac6120bfc/out/aesv8-armx-linux32.o" "/home/ikramulmurad/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/pregenerated/aesv8-armx-linux32.S"]: No such file or directory (os error 2)', /home/ikramulmurad/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/build.rs:653:9
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed

> Task :core:cargoBuildArm64 FAILED
   Compiling ring v0.16.20
   Compiling anyhow v1.0.42
   Compiling typemap v0.3.3
   Compiling getrandom v0.2.3
error: failed to run custom build command for `ring v0.16.20`

Caused by:
  process didn't exit successfully: `/home/ikramulmurad/Desktop/Workspace/_Projects/Android-Projects/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/debug/build/ring-120afe4acc70546b/build-script-build` (exit status: 101)
  --- stdout
  OPT_LEVEL = Some("0")
  TARGET = Some("aarch64-linux-android")
  HOST = Some("x86_64-unknown-linux-gnu")
  CC_aarch64-linux-android = None
  CC_aarch64_linux_android = None
  TARGET_CC = None
  CC = None
  CFLAGS_aarch64-linux-android = None
  CFLAGS_aarch64_linux_android = None
  TARGET_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("true")

  --- stderr
  running "aarch64-linux-android-clang" "-O0" "-DANDROID" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-I" "include" "-Wall" "-Wextra" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-g3" "-DNDEBUG" "-c" "-o/home/ikramulmurad/Desktop/Workspace/_Projects/Android-Projects/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/aarch64-linux-android/debug/build/ring-59f9fda9bc586338/out/aesv8-armx-linux64.o" "/home/ikramulmurad/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/pregenerated/aesv8-armx-linux64.S"
  thread 'main' panicked at 'failed to execute ["aarch64-linux-android-clang" "-O0" "-DANDROID" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-I" "include" "-Wall" "-Wextra" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-g3" "-DNDEBUG" "-c" "-o/home/ikramulmurad/Desktop/Workspace/_Projects/Android-Projects/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/aarch64-linux-android/debug/build/ring-59f9fda9bc586338/out/aesv8-armx-linux64.o" "/home/ikramulmurad/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/pregenerated/aesv8-armx-linux64.S"]: No such file or directory (os error 2)', /home/ikramulmurad/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/build.rs:653:9
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed

> Task :core:cargoBuildX86
   Compiling unicode-normalization v0.1.19
   Compiling ryu v1.0.5
   Compiling ring v0.16.20
   Compiling anyhow v1.0.42
error: failed to run custom build command for `ring v0.16.20`

Caused by:
  process didn't exit successfully: `/home/ikramulmurad/Desktop/Workspace/_Projects/Android-Projects/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/debug/build/ring-120afe4acc70546b/build-script-build` (exit status: 101)
  --- stdout
  OPT_LEVEL = Some("0")
  TARGET = Some("i686-linux-android")
  HOST = Some("x86_64-unknown-linux-gnu")
  CC_i686-linux-android = None
  CC_i686_linux_android = None
  TARGET_CC = None
  CC = None
  CFLAGS_i686-linux-android = None
  CFLAGS_i686_linux_android = None
  TARGET_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("true")

  --- stderr
  running "i686-linux-android-clang" "-O0" "-DANDROID" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-I" "include" "-Wall" "-Wextra" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-g3" "-DNDEBUG" "-c" "-o/home/ikramulmurad/Desktop/Workspace/_Projects/Android-Projects/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/i686-linux-android/debug/build/ring-316b98c54b0314a7/out/aesni-x86-elf.o" "/home/ikramulmurad/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/pregenerated/aesni-x86-elf.S"
  thread 'main' panicked at 'failed to execute ["i686-linux-android-clang" "-O0" "-DANDROID" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-I" "include" "-Wall" "-Wextra" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-g3" "-DNDEBUG" "-c" "-o/home/ikramulmurad/Desktop/Workspace/_Projects/Android-Projects/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/i686-linux-android/debug/build/ring-316b98c54b0314a7/out/aesni-x86-elf.o" "/home/ikramulmurad/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/pregenerated/aesni-x86-elf.S"]: No such file or directory (os error 2)', /home/ikramulmurad/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/build.rs:653:9
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed

> Task :core:cargoBuildX86 FAILED

> Task :core:cargoBuildX86_64 FAILED
   Compiling bitflags v1.2.1
   Compiling ring v0.16.20
   Compiling typemap v0.3.3
   Compiling anyhow v1.0.42
error: failed to run custom build command for `ring v0.16.20`

Caused by:
  process didn't exit successfully: `/home/ikramulmurad/Desktop/Workspace/_Projects/Android-Projects/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/debug/build/ring-120afe4acc70546b/build-script-build` (exit status: 101)
  --- stdout
  OPT_LEVEL = Some("0")
  TARGET = Some("x86_64-linux-android")
  HOST = Some("x86_64-unknown-linux-gnu")
  CC_x86_64-linux-android = None
  CC_x86_64_linux_android = None
  TARGET_CC = None
  CC = None
  CFLAGS_x86_64-linux-android = None
  CFLAGS_x86_64_linux_android = None
  TARGET_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("true")

  --- stderr
  running "x86_64-linux-android-clang" "-O0" "-DANDROID" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-I" "include" "-Wall" "-Wextra" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-g3" "-DNDEBUG" "-c" "-o/home/ikramulmurad/Desktop/Workspace/_Projects/Android-Projects/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/x86_64-linux-android/debug/build/ring-e0aa657d0b124c13/out/aesni-x86_64-elf.o" "/home/ikramulmurad/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/pregenerated/aesni-x86_64-elf.S"
  thread 'main' panicked at 'failed to execute ["x86_64-linux-android-clang" "-O0" "-DANDROID" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-I" "include" "-Wall" "-Wextra" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-g3" "-DNDEBUG" "-c" "-o/home/ikramulmurad/Desktop/Workspace/_Projects/Android-Projects/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/x86_64-linux-android/debug/build/ring-e0aa657d0b124c13/out/aesni-x86_64-elf.o" "/home/ikramulmurad/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/pregenerated/aesni-x86_64-elf.S"]: No such file or directory (os error 2)', /home/ikramulmurad/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/build.rs:653:9
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed

> Task :mobile:mergeDebugJniLibFolders UP-TO-DATE
> Task :mobile:validateSigningDebug UP-TO-DATE
> Task :mobile:writeDebugAppMetadata UP-TO-DATE
> Task :mobile:writeDebugSigningConfigVersions UP-TO-DATE

FAILURE: Build completed with 4 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':core:cargoBuildArm'.
> Process 'command 'cargo'' finished with non-zero exit value 101

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':core:cargoBuildArm64'.
> Process 'command 'cargo'' finished with non-zero exit value 101

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

3: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':core:cargoBuildX86'.
> Process 'command 'cargo'' finished with non-zero exit value 101

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

4: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':core:cargoBuildX86_64'.
> Process 'command 'cargo'' finished with non-zero exit value 101

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.3-rc-1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 17s
85 actionable tasks: 7 executed, 78 up-to-date

**Comment by Ravikiran-Bhonagiri** [comment]
+1 

I am facing the same issue.

**Comment by BastienVigneron** [comment]
+1

**Comment by lennartkloock** [comment]
[This](https://github.com/rust-lang/rust/issues/63939#issuecomment-665952545) solved the problem for me

**Reply by tyler-harpool** [reply]
Since there is no answer for Windows x64. 

Use WSL2 to run the build and it will succeed.

**Comment by Sulaiman001** [comment]
@tyler-harpool The  working solution for windows
1.	Install Visual Studio, make sure you select—.NET desktop development, Desktop development with C++, and Universal Windows Platform development.
2.	Show the installed toolchain `rustup show`.
3.	Then change and set the toolchain by running `rustup default stable-x86_64-pc-windows-msvc`.
This is what worked for me after weeks of not getting a working solution. I hope it work for someone else.

**Reply by circle-fish** [reply]
OMG！it  worked,thanks a lot

**Reply by taporyclub** [reply]
Best, Its worked for Windows x64. Thank You, i forgot about 'msvc' run on two list of Rust Show.

**Reply by sumonskys** [reply]
Thanks it work for me

**Reply by canntii** [reply]
Works for me!!

**Comment by TheFurnace** [comment]
Had this issue when using chocolatey to install rust, issue resolved by reinstalling using rustup.

**Comment by LongZoz** [comment]
I ran into a similar problem when compiling rust programs in ubuntu 18.04 (same results with gcc 7.5 and 9.4) and found that these packages were causing the problems: rbatis, mqtt, MongoDB; When I do not introduce them, there will be no error. I can not change the introduction of the lower version, but my program needs to use them. How can I deal with it

**Comment by lishuanwang** [comment]
***@***.***邮编：710069地址：陕西省西安市太白北路229号西北大学

**Comment by elibroftw** [comment]
In project directory: rustup override set stable-msvc
